client/data/site-roles/with-site-roles.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import useSiteRolesQuery from './use-site-roles-query';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+const withSiteRoles = createHigherOrderComponent(
```

Comment:
From my understanding it should be `withSiteRoles` because it provides `siteRoles` to the given component. I don't see an existing example with queries but e.g. [`withUpdateUser` ](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/people/edit-team-member-form/with-update-user.js) is named after the function we pass to the component as well. I don't feel strong about it though so am open to change it.

Reply:
My goal is to have a convention that's consistently used anywhere. And seems that we have it: hooks use the `*Query` or `*Mutation` suffix, and HOCs remove that suffix. I'm fine with that. Thanks for pointing out the existing `withUpdateUser` example.